### ISSUE TITLE
Proper CSM security module

### ISSUE THREAD

**Issue by ShadowNinja** [minetest/minetest#5958]
Currently CSM just uses the server's security functions -- but it should have its own security setup because the server security was made for backwards compatibility and is therefore a bit lenient and hacky<sup>[1]</sup>.  A module developed for the client doesn't have this concern and can provide better security.

This MUST be done before the CSM is stable!

[1]: For example, the secure path checking code is kind of a mess.  Client mods shouldn't have any direct filesystem access at all.  `secureLoadFile` is also pretty ugly and has been the source of a number of bugs.

**Comment by rubenwardy** [comment]
Duplicate of #7041
